pkg/segment/utils/number_test.go

```diff
@@ -0,0 +1,402 @@
+
+	floatVal, err := n.Float64()
+	assert.True(t, floatVal == 0, nil)
+	assert.True(t, err == nil)
+}
+
+// Test_isInt64 tests if the number is an int64. It should return set value for Int64 and an error for Float64.
+func Test_isInt64(t *testing.T) {
+
+	n := Number{}
+	n.SetInt64(42)
+	assert.True(t, n.ntype() == int64Type)
+
+	val, err := n.Int64()
+	assert.True(t, val == 42)
+	assert.True(t, err == nil)
+
+	_, err = n.Float64()
+	assert.True(t, err != nil)
+
+}
+
+// Test_isFloat64 tests if the number is a float64. It should return set value for Float64 and an error for Int64.
+func Test_isFloat64(t *testing.T) {
+
+	n := Number{}
+	n.SetFloat64(42.0)
+	assert.True(t, n.ntype() == float64Type)
+
+	val, err := n.Float64()
+	assert.True(t, val == 42.0)
+	assert.True(t, err == nil)
+
+	_, err = n.Int64()
+	assert.True(t, err != nil)
+
+}
+
+// Test_reset tests resetting the number. It should return invalidType.
+func Test_reset(t *testing.T) {
+
+	n := Number{}
+	n.SetInt64(42)
+	assert.True(t, n.ntype() == int64Type)
+	n.Reset()
+	assert.True(t, n.ntype() == invalidType)
+
+}
+
+// Test_constructor tests the constructor of the number. It should return the correct type.
+func Test_constructor(t *testing.T) {
+
+	n := Number{}
+	assert.True(t, n.ntype() == invalidType)
+
+	n = Number{[9]byte{0, 0, 0, 0, 0, 0, 0, 0, backfillType}}
+	assert.True(t, n.ntype() == backfillType)
+
+	n = Number{[9]byte{0, 0, 0, 0, 0, 0, 0, 0, int64Type}}
+	assert.True(t, n.ntype() == int64Type)
+
+	n = Number{[9]byte{0, 0, 0, 0, 0, 0, 0, 0, float64Type}}
+	assert.True(t, n.ntype() == float64Type)
+
+}
+
+// Test_numberSetAndGet tests setting and getting the number. It should return the correct value.
+func Test_numberSetAndGet(t *testing.T) {
+	n := Number{}
+
+	// Test Int64
+	n.SetInt64(42)
+	if val, err := n.Int64(); err != nil || val != 42 {
+		assert.Fail(t, "Expected 42, got %v with error %v", val, err)
+	}
+
+	// Test Float64
+	n.SetFloat64(3.14)
+	if val, err := n.Float64(); err != nil || math.Abs(val-3.14) > 1e-6 {
+		assert.Fail(t, "Expected 3.14, got %v with error %v", val, err)
+	}
+
+	// Test Invalid Type
+	n.SetInvalidType()
+	if _, err := n.Int64(); err == nil {
+		assert.Fail(t, "Expected error for invalid type")
+	}
+	if _, err := n.Float64(); err == nil {
+		assert.Fail(t, "Expected error for invalid type")
+	}
+
+	// Test Backfill Type
+	n.SetBackfillType()
+	if val, err := n.Int64(); err != nil || val != 0 {
+		assert.Fail(t, "Expected 0 for backfill, got %v with error %v", val, err)
+	}
+	if val, err := n.Float64(); err != nil || val != 0 {
+		assert.Fail(t, "Expected 0 for backfill, got %v with error %v", val, err)
+	}
+}
+
+// Test_numberConversion tests converting the number. It should return the correct value.
+func TestNumberConversion(t *testing.T) {
+	n := Number{}
+
+	// Test Int64 to Float64 conversion
+	n.SetInt64(42)
+	if err := n.ConvertToFloat64(); err != nil {
+		assert.Fail(t, "Conversion failed: %v", err)
+	}
```

Comment:
`assert` the type here: `ntype()` and then proceed to verify the `float` value. Sam comment for `float` to `int` conversion.

Reply:
Ahh I see you did the assertion before the conversion. That is fine. But I actually meant after the conversion. Do an assert on the type.

Reply:
Ohh yes right misunderstood it. Comparing the type after conversion.